SokolTextGame/Commands.cs

```diff
@@ -137,13 +166,60 @@ public void Execute(string[] words, World world)
                 Console.ForegroundColor = ConsoleColor.Green;
                 Random rand = new Random();
                 int randomIdex = rand.Next(0, locObj.LinesObjectSays.Length);
-                Console.WriteLine($"Guard: {locObj.LinesObjectSays[randomIdex]}");
+                Console.WriteLine($"{locObj.Name}: {locObj.LinesObjectSays[randomIdex]}");
                 Console.ResetColor();
             }
             else Console.WriteLine("There is no such thing here.");
             Console.ResetColor();
         }
     }
+    public class What : ICommand
+    {
+        public void Execute(string[] words, World world)
+        {
+            Console.ForegroundColor = ConsoleColor.Red;
+            var locObj = world.CurrentLocation().ObjectOnLocation;
+            if (string.Join(" ", words) == "what") Console.WriteLine("You can use the command \"What can I buy\"");
+            else if ((string.Join(" ", words) == "what can i buy"))
+            {
+                if (locObj?.ItemsForSale != null)
+                {
+                    var output = string.Join(", ", locObj.ItemsForSale);
+                    Console.ForegroundColor = ConsoleColor.Green;
+                    Console.WriteLine($"You can buy : {output} from a {locObj.Name}\n");
+                }
+                else Console.WriteLine("No one is selling anything here.");
+            }
+            else Console.WriteLine("Use the command \"What can I buy\"");
+            Console.ResetColor();
+        }
+    }
+    public class Buy : ICommand
+    {
+        public void Execute(string[] words, World world)
+        {
+            Console.ForegroundColor = ConsoleColor.Red;
+            if (string.Join(" ", words) == "buy" || words.Length != 4) Console.WriteLine("Use: buy <weapon> from <seller>");
+            else if (world.CurrentLocation().ObjectOnLocation != null)
+            {
+                var locObj = world.CurrentLocation().ObjectOnLocation;
+                var weaponName = words[1];
+                if (locObj.ItemsForSale != null &&
+                    words[0] == "buy" &&
+                    locObj.ItemsForSale.Contains(words[1]) &&
```

Comment:
The game should react to the case where you tried to buy an object that doesn't exist (where this `Contains()` call returns false). It should say something to the tune of `<seller> doesn't have any <object>s to sell you.`. Currently it just says the generic thing about pestering, which is not as fun as directly reacting to what you tried to do.